lib/messages.h

```diff
@@ -80,7 +81,7 @@ void msg_add_option_group(GOptionContext *ctx);
 
 #define msg_debug(desc, tag1, tags...) 						  \
 	do { 									  \
-	  if (G_UNLIKELY(debug_flag))             				  \
+	  if (G_UNLIKELY(debug_flag) || G_UNLIKELY(startup_debug_flag))             				  \
 	    msg_event_suppress_recursions_and_send(                               \
```

Comment:
Well, you only disable debug in case startup is set and debug as an option isnt. For that we would need to decouple options from the actual check whether debug is enabled.

But I am ok with how it is as long as the check is not scattered around and at least extracted into a function.

Also resetting log_stderr is not a good solution to disable debug.